src/controllers/Resources.js

```diff
@@ -0,0 +1,154 @@
+const Resource = require('../models/Resources')
+const { StatusCodes } = require('http-status-codes');
+const { title } = require('process');
+
+const getAllResources = async (req, res) => {
+    const page = parseInt(req.query.page) || 1;
+    const limit = parseInt(req.query.limit) || 10;
+
+    try {
+        const offset = (page - 1) * limit;
+        const totalResources = await Resource.countDocuments({});
+        const Resources = await Resource.find({})
+            .sort('createdAt')
+            .skip(offset)
+            .limit(limit);
+
+            res.status(StatusCodes.OK).json({ 
+                Resources, 
+                currentPage: page,
+                totalPages: Math.ceil(totalResources / limit),
+                totalItems: totalResources, 
+            });
+    } catch (error) {
+        console.error(error);
+        res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ msg: 'Internal server error' })
+    }
+};
+
+const getUserResources = async (req, res) => {
+    const page = parseInt(req.query.page) || 1;
+    const limit = parseInt(req.query.limit) || 10;
+
+    try {
+        const offset = (page - 1) * limit;
+        const totalResources = await Resource.countDocuments({ createdBy: req.user.userId });
+        const Resources = await Resource.find({ createdBy: req.user.userId })
+            .sort('createdAt')
+            .skip(offset)
+            .limit(limit);
+
+            res.status(StatusCodes.OK).json({ 
+                Resources, 
+                currentPage: page,
+                totalPages: Math.ceil(totalResources / limit),
+                totalItems: totalResources, 
+            });
+    } catch (error) {
+        console.error(error);
+        res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ msg: 'Internal server error' })
+    }
+};
+
+const getResources = async (req, res) => {
+    const { 
+        user: { userId }, 
+        params: { id: ResourcesId } 
+    } = req;
+
+    try {
+        const resource = await Resource.findOne({
+            _id: ResourcesId, createdBy: userId
+        });
+    
+        if (!resource) {
+            return res.status(StatusCodes.NOT_FOUND).json({ msg: `No Resources with id ${ResourcesId}` });
+        }
+    
+        res.status(StatusCodes.OK).json({ resources });
+        
+    } catch (error) {
+        console.error(error);
+        res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ msg: "Internal server error" });
+    }
+    
+};
+
+const createResources = async (req, res) => {
+
+    req.body.createdBy = req.user.userId;
+    const Resources = await Resource.create(req.body);
+    res.status(StatusCodes.CREATED).json({ Resources });
+};
+
+const updateResources = async (req, res) => {
+    const {
+        body: { topic, question, answer, createdBy },
```

Comment:
update line 86 to the resources fields: body: { title, topic, subtopic, type, link, createdBy },
(after you add the createdBy field to the model)